clients/unshaded/src/main/java/tachyon/client/lineage/TachyonLineageFileSystem.java

```diff
@@ -0,0 +1,172 @@
+   * @param outputFiles the files that the job outputs
+   * @param job the job to track
+   * @return the dependency id
+   * @throws IOException
+   * @throws FileDoesNotExistException
+   */
+  public long createLineage(List<TachyonURI> inputFiles, List<TachyonURI> outputFiles, Job job)
+      throws FileDoesNotExistException, IOException {
+    // TODO(yupeng): relax this to support other type of jobs
+    Preconditions.checkState(job instanceof CommandLineJob, "only command line job supported");
+
+    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      long lineageId = masterClient.createLineage(inputFiles, outputFiles, (CommandLineJob) job);
+      LOG.info("Created lineage " + lineageId);
+      return lineageId;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Deletes a lineage identified by a given id. If the delete is cascade, it will delete all the
+   * downstream lineages that depend on the given one recursively. Otherwise it does not delete the
+   * lineage exception, if there are other lineages whose input files are the output files of the
+   * specified lineage.
+   *
+   * @param lineageId the id of the lineage
+   * @param cascade whether to delete all the downstream lineages
+   * @return true if the lineage deletion is successful, false otherwise
+   * @throws IOException if the master cannot delete the lineage
+   * @throws LineageDeletionException if the deletion is cascade but the lineage has children
+   * @throws LineageDoesNotExistException if the lineage does not exist
+   */
+  public boolean deleteLineage(long lineageId, boolean cascade)
+      throws IOException, LineageDoesNotExistException, LineageDeletionException {
+    LineageMasterClient masterClient = mContext.acquireMasterClient();
+    try {
+      boolean result = masterClient.deleteLineage(lineageId, cascade);
+      LOG.info(result ? "Succeeded to " : "Failed to" + "delete lineage " + lineageId);
+      return result;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * Lists all the lineages.
+   *
+   * @return the informaiton about lineages
+   * @throws IOException if the master cannot list the lineage info
+   */
+  public List<LineageInfo> listLineages() throws IOException {
+    LineageMasterClient masterClient = mContext.acquireMasterClient();
+
+    try {
+      List<LineageInfo> result = masterClient.listLineages();
+      return result;
+    } finally {
+      mContext.releaseMasterClient(masterClient);
+    }
+  }
+
+  /**
+   * A file is created when its lineag is added. This method reinitializes the created file. But
+   * it's no-op if the file is already completed.
+   *
+   * @return the id of the reinitialized file when the file is lost or not completed, -1 otherwise.
+   * @throws LineageDoesNotExistException if the lineage does not exist.
+   */
+  public long recreate(TachyonURI path, OutStreamOptions options)
```

Comment:
If the file already exists, should it take a TachyonFile instead of path?

Also, should it return a TachyonFile instead of a long? I think we try to "hide" the id of the file in TachyonFile.

Also, when would a user need to call recreate()?